R/readGWLdata.R

```diff
@@ -10,50 +10,142 @@
+  if(which == "daily") which <- "average"
   
-  welldf <- read.csv(path, skip=5, row.names=NULL, stringsAsFactors=FALSE
-                     , na.strings=c("","N/A","NA","No Reading","NoReading"))
+  if(is.null(url)) url <- "http://www.env.gov.bc.ca/wsd/data_searches/obswell/map/data/"
   
-  # remove last row with html/java gibberish
-  welldf <- welldf[-nrow(welldf),]
+  if(!grepl("OW", well)) well <- paste0("OW", well)
   
-  names(welldf) <- c("Station_Name", "Date", "GWL", "Historical_Daily_Average"
-                     , "Historical_Daily_Minimum", "Historical_Daily_Maximum"
-                     , "Status")
+  url <- paste0(url, well, "-")
+  gwl <- download_gwl(url, which, quiet = quiet)
+  gwl <- format_gwl(gwl, quiet = quiet)
   
-  welldf <- welldf[!is.na(welldf$GWL),]
+  return(gwl)
+}
+
+#' Download groundwater data from url
+#' 
+#' @noRd
+download_gwl <- function(url, which, quiet) {
   
-  ## Extract Well number and if deep or shallow from station name:
-  st_name <- welldf$Station_Name[1]
-  wl_num <- gsub("OBS\\w*\\s*WELL\\s*#*\\s*(\\d+)(\\s.+|$)", "\\1", st_name)
-  if (grepl("shallow", st_name, ignore.case = TRUE)) wl_num <- paste(wl_num, "shallow")
-  if (grepl("deep", st_name, ignore.case = TRUE)) wl_num <- paste(wl_num, "deep")
+  if(!quiet) message("Retrieving data...")
+  gwl_data <- httr::GET(paste0(url, which, ".csv"))
+  httr::stop_for_status(gwl_data)
+  gwl_data <- httr::content(gwl_data, as = "text", encoding = "UTF-8")
+  
+  gwl_avg <- httr::GET(paste0(url, "minMaxMean.csv"))
+  httr::stop_for_status(gwl_avg)
+  gwl_avg <- httr::content(gwl_avg, as = "text", encoding = "UTF-8")
+  
+  return(list(gwl_data, gwl_avg))
+}
 
-  ## Set station name and well number
-  welldf$Station_Name <- st_name
-  welldf$Well_Num <- wl_num
+format_gwl <- function(data, quiet) {
+  
+  if(!quiet) message("Formatting data...")
+  
+  welldf <- utils::read.csv(text = data[[1]], stringsAsFactors = FALSE)
+  
+  welldf$myLocation <- gsub("OW", "", welldf$myLocation)
   
+  # For average data
+  if("QualifiedTime" %in% names(welldf)) {
+    welldf <- dplyr::rename(welldf, "Time" = "QualifiedTime")
+    welldf$Approval <- "Validated"
+  }
+  
+  # Merge with mean/min/max  
+  welldf$Time <- as.POSIXct(welldf$Time, tz="UTC")
+  welldf$dummydate <- paste0("1800-", format(welldf$Time, "%m-%d"))
+  
+  well_avg <- utils::read.csv(text = data[[2]], stringsAsFactors = FALSE)
+  well_avg <- well_avg[, names(well_avg)[names(well_avg) != "year"]]
+  well_avg <- tidyr::spread(well_avg, "type", "Value")
   
-  welldf$Date <- as.POSIXct(strptime(welldf$Date
-                                     , format="%Y-%m-%d %H:%M", tz="GMT"))
   
-  welldf$EMS_ID <- emsID
   
+  welldf <- dplyr::left_join(welldf, 
+                             well_avg[, c("dummydate", "max", "mean", "min")],
+                             by = "dummydate")
+  
+  ################################
+  # Need station name/location meta information!
```

Comment:
Absolutely, no hurry. We could still push to master, if you think people may want to use it without the meta data. Or we can wait if you think it could get confusing.